dbms/src/Flash/Coprocessor/DAGExpressionAnalyzer.cpp

```diff
@@ -588,22 +602,33 @@ void DAGExpressionAnalyzer::buildAggGroupBy(
             if (removeNullable(type)->isString())
                 collator = getCollatorFromExpr(expr);
             if (!duplicated_key)
-                collators.push_back(collator);
+                collators.back() = collator;
             if (collator != nullptr)
             {
-                /// if the column is a string with collation info, the `sort_key` of the column is used during
-                /// aggregation, but we can not reconstruct the origin column by `sort_key`, so add an extra
-                /// extra aggregation function any(group_by_column) here as the output of the group by column
-                TiDB::TiDBCollators arg_collators{collator};
-                appendAggDescription(
-                    {name},
-                    {type},
-                    arg_collators,
-                    "any",
-                    aggregate_descriptions,
-                    aggregated_columns,
-                    false,
-                    context);
+                auto [first_row_name, first_row_type] = findFirstRow(aggregate_descriptions, name);
```

Comment:
Is `first_row_type` necessary here? Are the `type` at L600 and `first_row_type` always the same? If so, can we just use `type`?

Reply:
I think they are same. And add a check to make sure.